src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,543 @@
+import { unified } from 'unified';
+import html from 'remark-html';
+import {
+  DEFAULT_EXPRESSION,
+  LEADING_HYPHEN,
+  NAME_EXPRESSION,
+  PARAM_EXPRESSION,
+  RETURN_EXPRESSION,
+  TYPE_EXPRESSION,
+} from '../constants.mjs';
+import { buildHierarchy } from './buildHierarchy.mjs';
+
+const sectionTypePlurals = {
+  module: 'modules',
+  misc: 'miscs',
+  class: 'classes',
+  method: 'methods',
+  property: 'properties',
+  global: 'globals',
+  example: 'examples',
+  // Constructors should go under a class sections' signatures property
+  ctor: 'signatures',
+  classMethod: 'classMethods',
+  event: 'events',
+  var: 'vars',
+};
+
+/**
+ * @param {import('../types.d.ts').HierarchizedEntry} entry
+ * @returns {import('../types.d.ts').Meta | undefined}
+ */
+function createMeta(entry) {
+  const makeArrayIfNotAlready = val => (Array.isArray(val) ? val : [val]);
+
+  const { added_in, n_api_version, deprecated_in, removed_in, changes } = entry;
+  if (added_in || n_api_version || deprecated_in || removed_in) {
```

Comment:
There is no need for this if statement. Let's diverge a bit from the API here and make it an empty array if the value is undefined.